### DISCUSSION TITLE
Add keyword filter | blacklist

### DISCUSSION THREAD

**Discussion by richardbenedikt** [jocmp/capyreader#475] [Feature Requests]
Thank you for your work on this fantastic app.

I would like to propose a new feature that could enhance user experience: Keyword Filtering.

It would be beneficial to have an option to blacklist content by keywords. This feature would allow users to input specific keywords in the settings, like "Advertisement" or "Championship," separated by commas. Articles containing these keywords (either in the title or the text) would be either excluded from display or not fetched at all.

**Comment by jocmp** [comment]
hey @richardbenedikt - thanks for the feature suggestion. I can't promise a timeline for this, but this seems like a reasonable addition. I know [Feeder](https://github.com/spacecowboy/Feeder) has a similar feature already.

**Reply by richardbenedikt** [reply]
Hey @jocmp, thanks for considering this suggestion! As you write, I think that this feature could be useful for many users. I personally found it very helpful in [Flym](https://github.com/FredJul/Flym), which I personally used extensively before it was discontinued.

**Reply by jocmp** [reply]
sure thing - I'll take another look at that implementation too. I think a few people have found Capy after coming from Flym.

**Comment by jocmp** [comment]
I've added a keyword filter feature for local feeds in [2025.03.1116-dev](https://github.com/jocmp/capyreader/releases/tag/2025.03.1116-dev). This matches on keywords in the title or content text across all feeds which is similar to Flym's implementation.

This can be found under Settings > General > "Blocklist".

<img src="https://github.com/user-attachments/assets/180627b1-a955-4f7d-a952-ab8cc832979d" width="300px">

**Reply by richardbenedikt** [reply]
Awesome, can't wait to try it out. Thank you!

**Reply by jaxaCe** [reply]
Hello, does the filter also match parts of the URL, I have only just discovered the feature and have not yet been able to try it out?

In my opinion, the filter should also be adjustable per feed.
For example, I would like to filter articles with a keyword in one feed which should not be filtered in the other feed.
Specifically, for example, I don't want to see anything to do with Android in one feed, but I do want to see this in an IT feed.

It should also be possible to specify where to search for keywords (URL, title, text).
This reduces the number of articles filtered as false positives for certain keywords.

A log of the filtered articles could be helpful to recognize false positives.
It would also be possible for an option to ensure that the articles are not actually deleted but simply marked as read, so that nothing can be lost.

**Reply by issteve** [reply]
A "nice to have" for me would be a group and switch possibility like in the Tusky App (Mastodon). So sometimes I need a healthy pause of certain topics (like war) and sometimes I'm strong enough to get informed again. Or I have more time to read and can switch to some normaly filtered topics for some time... 
Having the possibility to group some words (like Russia, Ukraine, Putin, Syria, Sudan, Somalia) under one topic e.g. "conflict" and a switch to turn this group on and off would help me to keep the information flood under healthier control... 
Thank you for your consideration.

**Reply by jocmp** [reply]
Hey @issteve - I definitely agree with you on taking a break from certain news topics, especially these days. I'm only intending on making this feature to be a global setting for now. I think advanced filtering is a good fit for a service like FreshRSS or Feedbin, but I'll take a look at Tusky's implementation!

**Reply by issteve** [reply]
Hey @jocmp  Thank you very much! I don't like and don't use any service provider like FreshRSS or Feedbin and use only direct website resources (I think here it's called "local " feeds). 
I was surprised as the blocklist didn't seem to work with new added expressions. I did some refresh and force closed/restart but the should be blocked article stayed there... But after removing the feed and adding it again the article disappeared. So it looks like it's blocking on the first fetch only. Fine for me, but maybe worth mentioning. 

I wonder if a kind of filter function to make articles automatically hidden or mark them read instead of really completely blocking them would be something nice to have... 
As there is already the filter bar for All/Unread/Favorites it could have a fourth option "auto" or "custom". 
Maybe this automatic filter function can also mark articles with certain words as favorites, too...

I'm very happy how Capy is already so again a big thank you for you anyway!

**Reply by jocmp** [reply]
All good ideas. I'm trying to keep this feature small at the moment, but I'll definitely keep these improvements in mind if I revise it! I added this to a issue here for future iteration https://github.com/jocmp/capyreader/issues/997
